indexer/blockapi.go

```diff
@@ -0,0 +1,169 @@
+package indexer
+
+import (
+	"context"
+	"errors"
+	"io"
+	"sync"
+
+	blocksapi "github.com/aurora-is-near/borealis-prototypes-go/generated/blocksapi"
+	"github.com/aurora-is-near/relayer2-base/broker"
+	"github.com/aurora-is-near/relayer2-base/db"
+	"github.com/aurora-is-near/relayer2-base/log"
+	"github.com/aurora-is-near/relayer2-base/types"
+	"github.com/aurora-is-near/relayer2-base/types/common"
+	"github.com/aurora-is-near/relayer2-base/types/indexer"
+	"github.com/aurora-is-near/relayer2-base/utils"
+	jsoniter "github.com/json-iterator/go"
+
+	vtgrpc "github.com/planetscale/vtprotobuf/codec/grpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/encoding"
+	_ "google.golang.org/grpc/encoding/proto"
+	"google.golang.org/grpc/metadata"
+)
+
+type IndexerBlocksAPI struct {
+	token  string
+	dbh    db.Handler
+	l      *log.Logger
+	b      broker.Broker
+	grpc   *grpc.ClientConn
+	height uint64
+	mu     sync.Mutex
+}
+
+func init() {
+	// Register protobuf plugin for generate optimized marshall & unmarshal code
+	encoding.RegisterCodec(vtgrpc.Codec{})
+}
+
```

Comment:
Also please add some logging so the progress of indexing is visible.
I would simply start second goroutine that would every couple of seconds log the current height, speed, etc
Something like I did here: https://github.com/aurora-is-near/block-storage/blob/main/cmd/testconsumer/testconsumer.go#L213
Indexer and this logging goroutine can exchange state via atomic vars easily